src/Interpreters/inplaceBlockConversions.cpp

```diff
@@ -337,9 +338,14 @@ void fillMissingColumns(
         }
         else
         {
-            /// We must turn a constant column into a full column because the interpreter could infer
-            /// that it is constant everywhere but in some blocks (from other parts) it can be a full column.
-            res_columns[i] = type->createColumnConstWithDefaultValue(num_rows)->convertToFullColumnIfConst();
+            if (requested_column->name == BlockNumberColumn.name)
+                res_columns[i] = type->createColumnConst(num_rows, block_number)->convertToFullColumnIfConst();
```

Comment:
Not sure, maybe we need to do it only for zero-level parts?

Reply:
`_block_number` column is only empty for zero-level parts